interpreter/transport/backend.go

```diff
@@ -0,0 +1,139 @@
+package transport
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/pkg/errors"
+	"github.com/ysugimoto/falco/ast"
+	"github.com/ysugimoto/falco/config"
+	"github.com/ysugimoto/falco/interpreter/context"
+	"github.com/ysugimoto/falco/interpreter/exception"
+	flchttp "github.com/ysugimoto/falco/interpreter/http"
+	"github.com/ysugimoto/falco/interpreter/value"
+)
+
+const (
+	HTTP_SCHEME  = "http"
+	HTTPS_SCHEME = "https"
+
+	DEFAULT_FIRST_BYTE_TIMEOUT_SEC = 15
+)
+
+// https://developer.fastly.com/reference/vcl/declarations/backend/
+type PropTypes interface {
+	*ast.String | *ast.Integer | *ast.RTime | *ast.Boolean | *ast.Ident
+}
+
+func getBackendProperty[T PropTypes](backend *value.Backend, key string) (T, error) {
+	var prop ast.Expression
+	for _, p := range backend.Value.Properties {
+		if p.Key.Value == key {
+			prop = p.Value
+			break
+		}
+	}
+	if prop == nil {
+		return nil, nil
+	}
+	if v, ok := prop.(T); ok {
+		return v, nil
+	}
+	return nil, errors.New("Type assertion failed")
+}
+
+func backendScheme(backend *value.Backend, override *config.OverrideBackend) (string, error) {
+	// scheme may be overrided by config
+	if override != nil {
```

Comment:
Since this pair of if statements are logically one thing it could be simplified into a single condition. `override != nil && override.SSL`.  There are a couple similar pairings in this file as well, lines 56/57 and 131/132.

Largely a style preference nit though.

Reply:
Yes, but we could make it clear the `override.SSL` would be compared with a `bool` type.